### PR TITLE
Array algoritm task solution

### PR DESCRIPTION
**Pull request by YanaBocharova**
Given an integer array sorted in non-decreasing order, 
  there is exactly one integer in the array that occurs more than 25% of the time, return that integer.

### CODE REVIEW

**Comment by chocolatapie** [comment]
Hi

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AlgoritmicalTasksAndUT
+{
+    public class ArrayNumbers
+    {
+
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs:10]: Remove empty line

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                {
+                    if (dictNums.ContainsKey(arr[i]))
+                    {
+                        dictNums[arr[i]]++;
+                    }
+                    else
+                    {
+                        dictNums[arr[i]] = 1;
+                    }
+
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs:27]: remove empty line

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                    }
+                   
+                }
+                int maxNumber = dictNums.Values.Max();
+
+                foreach(var key in dictNums.Keys)
+                {
+                    if (dictNums[key] == maxNumber)
+                    {
+                        foundValue = key;
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs:35]: add `break` after this statement
we dont need to continue our loop, once we found an answer

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/Program.cs [hunk 4]
```diff
@@ -7,6 +7,23 @@ class Program
...
+
+            ArrayNumbers array = new ArrayNumbers();
+
+            int foundNumber=array.ReturnNumber(arr);
+
+            Console.WriteLine($"found integer= {foundNumber}");
+
+
+
+
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/Program.cs:25]: Extract this in a separate `private void` method, and remove empty lines

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksTests/ArrayNumbersTest.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            // arrange 
+            var ArrayNumTestObj = new ArrayNumbers();
+
+            // act
+            var result = ArrayNumTestObj.ReturnNumber(null);
+
+            // assert
+            Assert.AreEqual(-1, result);
+        }
+
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksTests/ArrayNumbersTest.cs:34]: remove empty line

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs (hunk 1 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs]: Fixed

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs (hunk 2 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs]: Fixed

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs (hunk 3 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/ArrayNumbers.cs:35]: Fixed

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/Program.cs (hunk 4 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/Program.cs]: Fixed

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksTests/ArrayNumbersTest.cs (hunk 5 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksTests/ArrayNumbersTest.cs:34]: Fixed

**Review by chocolatapie — APPROVED** [review]

**Pull request merged**